### PR TITLE
bugfix: Fix document highlight in for comprehensions

### PR DESCRIPTION
**Pull request by tgodzik**
Previously, document highlight in for comprehensions would not highlight both the definition and the usages in yield. This was caused by the fact that these two are really just separate symbols in separate anonymous functions. Now, we try to relate the two owner functions based on their point position, which seems to work correctly.

It's a bit different in case of Scala 3, since there seem to be more trees with the particular symbol generated, but still path will point us to different symbol than the one being used.

### CODE REVIEW

File: mtags/src/main/scala-3/scala/meta/internal/pc/PcDocumentHighlightProvider.scala [hunk 1]
```diff
@@ -262,7 +274,8 @@ object PcDocumentHighlightProvider:
 
         val traverser = new DeepFolder[Set[DocumentHighlight]](collectNames)
         val all = traverser(Set.empty[DocumentHighlight], unit.tpdTree)
-        all.toList.distinct
+
+        all.toList.distinctBy(_.getRange())
```
Comment (tgodzik) [inline mtags/src/main/scala-3/scala/meta/internal/pc/PcDocumentHighlightProvider.scala:278]: Looks like Scala 3 has several trees corresponding to the same position, but this workaround seems to work ok.

File: mtags/src/main/scala-2/scala/meta/internal/pc/PcDocumentHighlightProvider.scala [hunk 2]
```diff
@@ -158,6 +158,16 @@ final class PcDocumentHighlightProvider(
           .filter(_ != NoSymbol)
         lazy val soughtNames: Set[Name] = sought.map(_.name)
 
+        /*
+         * For comprehsnions have two owners, one for the enumerators and one for
```
Comment (kpodsiad) [inline mtags/src/main/scala-2/scala/meta/internal/pc/PcDocumentHighlightProvider.scala:162]:
```suggestion
         * For comprehensions have two owners, one for the enumerators and one for
```

File: mtags/src/main/scala-3/scala/meta/internal/pc/PcDocumentHighlightProvider.scala [hunk 3]
```diff
@@ -153,6 +153,17 @@ object PcDocumentHighlightProvider:
           .flatMap { s => Set(s.owner, s.owner.companionModule) }
           .filter(_ != NoSymbol)
         lazy val soughtNames: Set[Name] = sought.map(_.name)
+
+        /*
+         * For comprehsnions have two owners, one for the enumerators and one for
```
Comment (kpodsiad) [inline mtags/src/main/scala-3/scala/meta/internal/pc/PcDocumentHighlightProvider.scala:158]:
```suggestion
         * For comprehensions have two owners, one for the enumerators and one for
```

**Review by tanishiking — APPROVED** [review]
nice :+1:

**Pull request merged**